integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/utils/JMSBroker.java

```diff
@@ -151,4 +142,49 @@ public boolean isBrokerStarted() {
     private void setBrokerStatus(boolean status) {
         isBrokerStarted = status;
     }
-}
\ No newline at end of file
+
+    /**
+     * Starting ActiveMQ embedded broke after specifying plugins.
+     *
+     * @param brokerPlugins the array of BrokerPlugins to be set
+     * @return whether or not the broker was stated successfully
+     */
+    public boolean setPluginsAndStart(BrokerPlugin[] brokerPlugins) {
+        try {
+            setInitialConfigurations();
+            broker.setPlugins(brokerPlugins);
+            startBroker();
+            return true;
+        } catch (Exception e) {
+            log.error("JMSServerController: There was an error starting JMS broker: " + serverName, e);
+            return false;
+        }
+    }
+
+    /**
+     * Helper method to set initial broker configurations.
+     *
+     * @throws Exception if there is an error setting TransportConnectors
+     */
+    private void setInitialConfigurations() throws Exception {
+        log.info("JMSServerController: Preparing to start JMS Broker: " + serverName);
+        broker = new BrokerService();
+        broker.setBrokerName(serverName);
+        log.info(broker.getBrokerDataDirectory());
+        broker.setDataDirectory(System.getProperty(FrameworkConstants.CARBON_HOME) + File.separator
+                + broker.getBrokerDataDirectory());
+        broker.setTransportConnectors(transportConnectors);
+        broker.setPersistent(true);
+    }
+
+    /**
+     * Private method to start the broker.
+     *
+     * @throws Exception if there is an error starting the broker
+     */
+    private void startBroker() throws Exception {
```

Comment:
do we need this method ? can we set brokerState at the start() method and get rid of this ?

Reply:
The start() method here is that of the BrokerService provided by ActiveMQ, so we have to set the broker status here.